Show filename of notebook with failing cell

Example files in https://github.com/computationalmodelling/nbval/tree/master/issues/67

Issues is that 

When running

  `py.test --nbval notebook1.ipynb notebook2.ipynb`

the output is::

    ======================================= test session starts ===============
    platform darwin -- Python 3.6.0, pytest-3.0.7, py-1.4.33, pluggy-0.4.0
    rootdir: /Users/fangohr/git/nbval, inifile:
    plugins: nbval-0.6
    collected 2 items

    notebook1.ipynb .
    notebook2.ipynb F

    ============================================ FAILURES =====================
    _____________________________________________ cell 0 ______________________
    Notebook cell execution failed
    Cell 0: Cell outputs differ

    Input:
    import time

    time.time()

    Traceback: mismatch 'text/plain'
    <<<<<<<<<<<< Reference output from ipynb file:
    1498639847.528011
    ============ disagrees with newly computed (test) output:
    1498640219.550588
    >>>>>>>>>>>>
    =============================== 1 failed, 1 passed in 2.07 seconds ========

This is not ideal as we know there is a fail in Cell 0, but  we don't
know in which file. (Here it is ``notebook1.ipynb``).

Suggest that the name of the file in which the reported failures take
place is displayed somehow.

How could we add the information?

For failing tests in normal python files, the file name appears to be displayed at the end of the failure report (here `test_2.py`). Output produced with `py.test test*py` in https://github.com/computationalmodelling/nbval/tree/master/issues/67):

<pre>
============================================ FAILURES =============================================
_____________________________________________ test_b ______________________________________________

    def test_b():
>       raise AssertionError
E       AssertionError

test_2.py:2: AssertionError
_____________________________________________ test_c ______________________________________________

    def test_c():
>       assert 1 == 2
E       assert 1 == 2

test_2.py:5: AssertionError
=============================== 2 failed, 1 passed in 0.09 seconds ================================

</pre>

A possible analog would be to change the output (from `py.test --nbval *ipynb`) 
<pre>
============================================ FAILURES =============================================
_____________________________________________ cell 0 ______________________________________________
Notebook cell execution failed
Cell 0: Cell outputs differ

Input:
import time

time.time()

Traceback: mismatch 'text/plain'
<<<<<<<<<<<< Reference output from ipynb file:
1498639847.528011
============ disagrees with newly computed (test) output:
1498641462.5700061
>>>>>>>>>>>>
=============================== 1 failed, 1 passed in 4.09 seconds ================================
</pre>
to become (new second last line)
<pre>
============================================ FAILURES =============================================
_____________________________________________ cell 0 ______________________________________________
Notebook cell execution failed
Cell 0: Cell outputs differ

Input:
import time

time.time()

Traceback: mismatch 'text/plain'
<<<<<<<<<<<< Reference output from ipynb file:
1498639847.528011
============ disagrees with newly computed (test) output:
1498641462.5700061
>>>>>>>>>>>>
notebook2.ipynb:cell 0:
=============================== 1 failed, 1 passed in 4.09 seconds ================================
</pre>

We could also provide (redundant information) by also changing the 3rd line and adding the name of the notebook file in there as well. For example:
<pre>
============================================ FAILURES =============================================
_____________________________________________ cell 0 ______________________________________________
notebook2.ipynb: Notebook cell execution failed 
Cell 0: Cell outputs differ

Input:
import time

time.time()

Traceback: mismatch 'text/plain'
<<<<<<<<<<<< Reference output from ipynb file:
1498639847.528011
============ disagrees with newly computed (test) output:
1498641462.5700061
>>>>>>>>>>>>
notebook2.ipynb:cell 0:
=============================== 1 failed, 1 passed in 4.09 seconds ================================
</pre>

Thoughts?

This gives headers like this:
```
================================== FAILURES ===================================
_________________ test-latex-fail-randomoutput.ipynb::Cell 2 __________________
```

```
================================== FAILURES ===================================
___________________________ notebook2.ipynb::Cell 0 ___________________________
Notebook cell execution failed
Cell 0: Cell outputs differ

Input:
import time

time.time()

Traceback: mismatch 'text/plain'
<<<<<<<<<<<< Reference output from ipynb file:
1498639847.528011
============ disagrees with newly computed (test) output:
1498643832.155805
>>>>>>>>>>>>
===================== 1 failed, 1 passed in 4.24 seconds ======================
```

--- a/nbval/nbdime_reporter.py
+++ b/nbval/nbdime_reporter.py
@@ -143,7 +143,7 @@ class NbdimeReporter:
 
     def _getfailureheadline(self, rep):
         if hasattr(rep, 'location'):
-            fspath, lineno, domain = rep.location
-            return fspath + '::' + domain
+            domain = rep.location[2]
+            return domain
         else:
             return "test session"  # XXX?

--- a/nbval/plugin.py
+++ b/nbval/plugin.py
@@ -10,6 +10,7 @@ from __future__ import print_function
 # import the pytest API
 import pytest
 import sys
+import os
 import re
 import hashlib
 import warnings
@@ -348,7 +349,7 @@ class IPyNbCell(pytest.Item):
             return "pytest plugin exception: %s" % str(exc)
 
     def reportinfo(self):
-        description = "cell %d" % self.cell_num
+        description = "%s::Cell %d" % (self.fspath.relto(self.config.rootdir), self.cell_num)
         return self.fspath, 0, description
 
     def compare_outputs(self, test, ref, skip_compare=None):

